File: 581.最短无序连续子数组.py
#
# @lc app=leetcode.cn id=581 lang=python
#
# [581] 最短无序连续子数组
#

# @lc code=start
class Solution(object):
    def findUnsortedSubarray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        num = sorted(nums)
        sum = 0
        for i in range(len(nums)): 
            if num[i] != nums[i]:
                sum += 1
        return sum
        
# @lc code=end

File: 581.最短无序连续子数组.cpp
/*
 * @lc app=leetcode.cn id=581 lang=cpp
 *
 * [581] 最短无序连续子数组
 */

// @lc code=start
class Solution {
public:
    int findUnsortedSubarray(vector<int>& nums) {
        int start = 0, end = nums.size()-1;
        vector<int>com = nums;
        sort(com.begin(), com.end());
        while(start < end and com[start] == nums[start])
            start++;
        while(start < end and com[end] == nums[end])
            end--;
        
        return start == end ? 0 : end - start + 1;
    }
};
// @lc code=end

